content/blog/open-sourcing-workshops-on-demand-part3-understanding-the-backend.md

```diff
@@ -0,0 +1,340 @@
+
+T﻿his completion ratio script provides us this data and we store it in our database. 
+
+1﻿1- API call to send the completion ratio figure to the database. This can be later queried to build up nice reporting dashboard as explained here by my colleague Didier Lalli in the following [article](https://developer.hpe.com/blog/open-source-elasticsearch-helped-us-globally-support-virtual-labs/).
+
+1﻿2- `erase-student()`: now that we have extracted the necessary data from the student's notebooks, we can perform a cleanup of the student folder.
+
+1﻿3- `cleanup_processes_student()`: On top of cleanup of the student folder, we also kill all the allocated student's processes.
+
+1﻿4- `cleanup-<workshop>.sh`: If any appliance is involved, this task will perform the necessary cleanup processes on the appliance.
+
+1﻿5- Finally, just like for the creation of a workshop process, we need to inform back the frontend that now cleanup is done. Therefore, several API calls are made to update tables in the database. The new s﻿tudent password is recorded. We also generate a new password at the cleanup phase to prevent unregistered logins. The s﻿tudent status is set to inactive. The capacity figure is incremented by one to make the seat available again. 
+
+A﻿s for the **CREATE** phase, the regular checks occuring on the frontent web portal will get these data and trigger the final email to the participant thanking him for his particpation.
+
+N﻿ow let's look at the **RESET** scenario.
+
+![](/img/wod-blogserie3-reset.png "backend server RESET workflow")
+
+ You may wonder what are the differences between **CLEANUP** and **RESET**? Well, firstly, they spell differently but that has nothing to do with the purpose of this article...Secondly, **CLEANUP** only takes care of student whereas **RESET** takes care of a larger scope. Let me explain.
+
+W﻿hen a **CLEANUP** occurs, it deals with the participant's student workshop and home directory (the workshop directory belonging to the home directory). It cleans up workshop content, ssh keys, skeletons. The <RESET> will delete leftovers from the workshop's exercices. For intance, when one runs the [Kubernetes 101](https://developer.hpe.com/hackshack/workshop/24)workshop, he is creating microservices, he's scaling them, and should at the end of the workshop run some delete commands to clean up everything. However, this does happen all the time. And the admin needs to make sure that the next participant who will get affected the very same student environment comes with a fresh one. Therefore, some measures have to be taken. Well these measures take place when a reset flag is associated to the workshop in the database.
```

Comment:
For some reason, in the preview view, <RESET> did not show up and it just reads:
![image](https://user-images.githubusercontent.com/77496261/222981742-b85f437f-a83d-4d58-a2a7-f3bb2b2645ff.png)


So it looks like a word is missing where I have it highlighted. Please also note that exercises and instance have typos in them.

Right where it says "However, this does happen all the time." it feels to me that something else is missing. Read this paragraph carefully.  I'm pretty sure that the next section should read:
"And the admin needs to make sure that the next participant who will be assigned the same student environment gets a fresh one."

For the last sentence, take off the word "Well" and just start with "These measures take place..."

Reply:
done  replaced with However, some participants may forget to  run these clean up steps.